internal/pkg/cli/pipeline_init.go

```diff
@@ -166,26 +170,35 @@ func (o *initPipelineOpts) Validate() error {
 
 // Ask prompts for required fields that are not passed in and validates them.
 func (o *initPipelineOpts) Ask() error {
-	if o.repoURL != "" {
-		if err := o.validateURL(o.repoURL); err != nil {
-			return err
-		}
-	} else {
-		if err := o.selectURL(); err != nil {
+	if o.name == "" {
+		if err := o.askPipelineName(); err != nil {
 			return err
 		}
 	}
+	if err := validatePipelineName(o.name, o.appName); err != nil {
+		return err
+	}
 
-	if len(o.environments) > 0 {
-		if err := o.validateEnvs(); err != nil {
+	// TODO: validate not a duplicate
+
+	if o.repoURL == "" {
+		if err := o.selectURL(); err != nil {
 			return err
 		}
+	}
+	if err := o.validateURL(o.repoURL); err != nil {
+		return err
+	}
 
-	} else {
+	if len(o.environments) == 0 {
```

Comment:
Thank you for flipping this! Much better. 🎉

Reply:
We do in this case, at least to keep the same behavior as before. I removed the validation code from `askEnvs()`, since it already was written in `validateEnvs`